### PR TITLE
fix(explorer): fix explorer bugs

### PR DESCRIPTION
**Pull request by pavelnikolov**
* properly handle nil receipts when displaying txHash and receipt timestamp
* move block timestamp to source chain and not dest chain to fix confusion

task: none

### CODE REVIEW

File: explorer/ui/app/components/details/From.tsx [hunk 1]
```diff
@@ -65,7 +66,18 @@ export const From = ({ xMsgDetails }) => {
...
+        <p className="w-[150px] sm:w-48 text-sm">Block Timestamp</p>
+        <p className="flex-1">
+          <span className="text-default">
+            {dateFormatter(new Date(xMsgDetails?.block.timestamp))}{' '}
+          </span>
+         ( {xMsgDetails?.block.timestamp &&
+            dateFormatterXMsgPage(new Date(xMsgDetails?.block.timestamp))})
+        </p>
+      </div>
+      {/* Block height */}
```
Comment (pavelnikolov) [inline explorer/ui/app/components/details/From.tsx:80]: Move block timestamp to source chain to be next to the other block properties.

File: explorer/ui/app/components/details/From.tsx [hunk 2]
```diff
@@ -65,7 +66,18 @@ export const From = ({ xMsgDetails }) => {
...
-      {/* Offset */}
+      {/* Block Timestamp */}
+      <div className="flex mt-5 pb-2 border-b-[1px] border-subtle border-solid">
+        <p className="w-[150px] sm:w-48 text-sm">Block Timestamp</p>
+        <p className="flex-1">
+          <span className="text-default">
+            {dateFormatter(new Date(xMsgDetails?.block.timestamp))}{' '}
+          </span>
+         ( {xMsgDetails?.block.timestamp &&
+            dateFormatterXMsgPage(new Date(xMsgDetails?.block.timestamp))})
```
Comment (kevinhalliday) [inline explorer/ui/app/components/details/From.tsx:77]: Once you check for undefefined / null, you dont' have to check again

File: explorer/ui/app/components/details/TabList.tsx [hunk 3]
```diff
@@ -12,13 +12,17 @@ export const TabList = ({ xMsgDetails }) => {
...
+          <>
+            <span className="text-default">
+              {dateFormatter(new Date(xMsgDetails?.receipt?.timestamp))}{' '}
+            </span>
+            (
+            {xMsgDetails?.receipt?.timestamp &&
+              dateFormatterXMsgPage(new Date(xMsgDetails?.receipt?.timestamp))}
+            )
+          </>
+          ) : '--'}
```
Comment (pavelnikolov) [inline explorer/ui/app/components/details/TabList.tsx:25]: If receipt doesn't exist don't display anything.

File: explorer/ui/app/components/details/TabList.tsx [hunk 4]
```diff
@@ -55,22 +59,18 @@ export const TabList = ({ xMsgDetails }) => {
...
-      {/* Offset */}
       <div className="flex mt-5 pb-2 border-b-[1px] border-subtle border-solid">
         <p className="w-[150px] sm:w-48 text-sm">Stream Offset</p>
         <p className="text-default">{parseInt(xMsgDetails?.offset)}</p>
       </div>
-      {/* Offset */}
       <div className="flex mt-5 pb-2 border-b-[1px] border-subtle border-solid">
         <p className="w-[150px] sm:w-48 text-sm">Gas Used</p>
-        <p className="text-default">{parseInt(xMsgDetails?.receipt?.gasUsed)}</p>
+        <p className="text-default">{xMsgDetails?.receipt ? parseInt(xMsgDetails?.receipt?.gasUsed) : '--'}</p>
```
Comment (pavelnikolov) [inline explorer/ui/app/components/details/TabList.tsx:68]: Don't show `NaN` if receipt doesn't exist. Instead show `--` as we do with other fields in this case.

File: explorer/ui/app/components/details/From.tsx (hunk 2 above)
Comment (kevinhalliday) [reply explorer/ui/app/components/details/From.tsx:77]:
```suggestion
         ( {xMsgDetails?.block.timestamp &&
            dateFormatterXMsgPage(new Date(xMsgDetails.block.timestamp))})
```

Comment (kevinhalliday) [reply explorer/ui/app/components/details/From.tsx:77]: Also nit but I tend to[ prefer ternaries over &&](https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx)
```suggestion
         ( {xMsgDetails?.block.timestamp ? 
            dateFormatterXMsgPage(new Date(xMsgDetails?.block.timestamp)) : null})
```

**Review by kevinhalliday — APPROVED** [review]

File: explorer/ui/app/components/details/TabList.tsx (hunk 3 above)
Comment (kevinhalliday) [reply explorer/ui/app/components/details/TabList.tsx:25]: We'll, we do display `--` : could say something like "No Receipt", or actually nothing. Ask Fab

Comment (pavelnikolov) [reply explorer/ui/app/components/details/TabList.tsx:24]: This is what we do for tx_hash already for receipts. I copied that behavior. Happy to change it if required.

**Pull request merged**